homeassistant/components/climate/dyson.py

```diff
@@ -0,0 +1,194 @@
+    ClimateDevice, STATE_HEAT, STATE_COOL, STATE_IDLE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_FAN_MODE, SUPPORT_OPERATION_MODE)
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_DIFFUSE = "Diffuse Mode"
+STATE_FOCUS = "Focus Mode"
+FAN_LIST = [STATE_FOCUS, STATE_DIFFUSE]
+OPERATION_LIST = [STATE_HEAT, STATE_COOL]
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+                 | SUPPORT_OPERATION_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Dyson fan components."""
+    if discovery_info is None:
+        return
+
+    from libpurecoollink.dyson_pure_hotcool_link import DysonPureHotCoolLink
+    # Get Dyson Devices from parent component.
+    add_devices(
+        [DysonPureHotCoolLinkDevice(device)
+         for device in hass.data[DYSON_DEVICES]
+         if isinstance(device, DysonPureHotCoolLink)]
+    )
+
+
+class DysonPureHotCoolLinkDevice(ClimateDevice):
+    """Representation of a Dyson climate fan."""
+
+    def __init__(self, device):
+        """Initialize the fan."""
+        self._device = device
+        self._current_temp = None
+        self._pending_target_temp = None
+
+    async def async_added_to_hass(self):
+        """Callback when entity is added to hass."""
+        self.hass.async_add_job(self._device.add_message_listener,
+                                self.on_message)
+
+    def on_message(self, message):
+        """Called when new messages received from the climate."""
+        from libpurecoollink.dyson_pure_state import DysonPureHotCoolState
+
+        if isinstance(message, DysonPureHotCoolState):
+            from libpurecoollink.const import HeatMode
+            _LOGGER.debug("Message received for fan device %s : %s", self.name,
+                          message)
+            self.schedule_update_ha_state()
+            if (self._device.state.heat_mode == HeatMode.HEAT_ON.value
+                    and self._pending_target_temp is not None):
+                # Set any pending new target temperature on dyson device.
+                self._set_temperature_on_device()
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def name(self):
+        """Return the display name of this climate."""
+        return self._device.name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._device.environmental_state:
+            temperature_kelvin = self._device.environmental_state.temperature
+            if temperature_kelvin != 0:
+                self._current_temp = float("{0:.1f}".format(
```

Comment:
Maybe we should move it to `util/temperature.py`

Reply:
Yes maybe there should be a Kelvin conversion in `util/temperature.py`. But regardlessly, this should remains as the actual Kelvin conversion is something similar to `Celsius - 273.15`, but the device uses whole number substraction (i.e. `C - 273` internally).